### PR TITLE
adding dockstore vignette, in progress

### PR DESCRIPTION
**Pull request by vjcitn**
for this to work i needed to add auth.json to inst/service/dockstore

### CODE REVIEW

File: vignettes/biocDockstore.Rmd [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    toc: yes
+    number_sections: yes
+---
+
+# Introduction: Basic concepts of Dockstore and Bioconductor
+
+```{r setup,echo=FALSE}
+suppressPackageStartupMessages({
+library(AnVIL) # get from https://github.com/Bioconductor/AnVIL
+library(rjson)
```
Comment (mtmorgan) [inline vignettes/biocDockstore.Rmd]: We use the jsonlite package.

File: vignettes/biocDockstore.Rmd [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+library(httr)
```
Comment (mtmorgan) [inline vignettes/biocDockstore.Rmd]: Maybe httr and tidyverse (which is the paradigm I would like to adopt for end-users) should be added to the Depends: field of the DESCRIPTION file?

File: vignettes/biocDockstore.Rmd [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+```
+
+An object called 'dockstore' is defined and will be
+used to process API requests.  We'll define a little
+bit of temporary infrastructure to show what 'dockstore'
+can do.
+
+```{r getmeths}
+api = function(x) slot(x, "api")
+sapply(api(dockstore)$tags, "[[", "name")
```
Comment (mtmorgan) [inline vignettes/biocDockstore.Rmd]: `tags(dockstore) %>% distinct(tag)`

File: vignettes/biocDockstore.Rmd [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+```
+
+These are groups of API components.  We're interested in
+the 'users' component.  Higher level methods will be
+introduced to help here, but for now we stick to base R
+methods.
+
+```{r lku}
+umeth = sort(grep("[uU]ser", names(operations(dockstore)), value=TRUE))
```
Comment (mtmorgan) [inline vignettes/biocDockstore.Rmd]: Maybe `tags(dockstore) %>% filter(tag == "users") %>% pull(operation)` ?

File: vignettes/biocDockstore.Rmd [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+decode = function(x) rjson::fromJSON(httr::content(x, "text"))
+umeth
+```
+
+We can use the following to determine our user identifier.
+```{r lklk}
+myuid = dockstore$getUser()
+decode(myuid)$id
```
Comment (mtmorgan) [inline vignettes/biocDockstore.Rmd]: Maybe there should be an as.list.response method
```
as.list.response <-
    function(x, ...)
{
    as.list(content(x))
}
```
then `(dockstore$getUser() %>% as.list())$id`

File: vignettes/biocDockstore.Rmd (hunk 2 above)
Comment (vjcitn) [reply vignettes/biocDockstore.Rmd]: Depending on tidyverse seems to me unwarranted.  tidyverse cannot be loaded without readxl and haven ... broom, etc.  tidyverse_deps() returns 25 packages.  Certain dplyr/magrittr patterns seem ok to me but I would propose import as needed

File: vignettes/biocDockstore.Rmd (hunk 5 above)
Comment (vjcitn) [reply vignettes/biocDockstore.Rmd]: done

**Comment by vjcitn** [comment]
All suggestions adopted with exception of tidyverse.

**Pull request merged**